Library/Homebrew/sandbox.rb

```diff
@@ -68,6 +69,11 @@ def allow_write_cellar(formula)
     allow_write_path formula.var
   end
 
+  # Xcode projects expect access to certain cache/archive dirs.
+  def allow_write_xcode
+    allow_write_path "/Users/#{ENV["USER"]}/Library/Developer/Xcode/DerivedData/"
```

Comment:
Given we have an `xcodebuild` helper it'd be cool if we could integrate using that?